src/Microsoft.OData.Edm/Schema/Interfaces/IEdmTargetPath.cs

```diff
@@ -0,0 +1,22 @@
+//---------------------------------------------------------------------
+// <copyright file="IEdmTargetPath.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.OData.Edm.Vocabularies;
+
+namespace Microsoft.OData.Edm
+{
+    /// <summary>
+    /// Represents a path to a model element.
+    /// </summary>
+    public interface IEdmTargetPath : IEdmVocabularyAnnotatable
+    {
+        /// <summary>
+        /// Gets the target path segments as Edm elements.
+        /// </summary>
+        IEnumerable<IEdmElement> Segments { get; }

```

Comment:
>IEnumerable [](http://example.com/codeflow?start=8&length=11)

Maybe it's better to use other generic collection?

Reply:
Will leave it as `IEnumerable<IEdmElement>`.
Internally we will be manipulating a `List<IEdmElement>`

Reply:
@KenitoInc I believe that is the right call...

Reply:
@KenitoInc ☝️ Not a bad idea